Configure

```diff
@@ -1662,12 +1662,16 @@ if ($shlib_version_number =~ /(^[0-9]*)\.([0-9\.]*)/)
 	}
 
 my $ecc = $cc;
-$ecc = "clang" if `$cc --version 2>&1` =~ /clang/;
+{
+my $ccv = `$cross_compile_prefix$cc --version 2>&1`;
+$ecc = "clang" if $ccv =~ /clang/;
+$ecc = "gcc" if $ccv =~ /gcc/;
```

Comment:
On side note it was pointed out that gcc compiler driver reports basename of argv[0] in --version output. How is it a problem? Customarily you'd have /usr/bin/cc that is gcc in disguise, but this check won't recognize it as gcc as it will report `cc 3.2.1`. Also note that archaic gcc versions print just version number, e.g. 2.25.3, i.e. don't identify themselves the expected way. None of this applies to clang, so that check for presence of "clang" in --version output is reliable. Unlike gcc that is.

Reply:
It should be noted however that as far as MAKEDEPPROG goes that thing that archaic gcc versions print just version number incidentally works out. This is because archaic versions don't support generation of dependencies, and you don't "want" to recognize it as gcc :-)

Reply:
Another problem with check for gcc in --version output is ... IBM compiler. When it doesn't recognize command line option, such as --version here, it emits manual page, which mentions gcc, so that the check should come out successful...